### PR TITLE
Value-based tests for log-likelihoods

### PR DESCRIPTION
**Pull request by ben18785**
In this checking of the tests, I (somewhat disconcertingly) found about a dozen errors that would have been picked up if we were doing value-based tests of log-likelihoods. The errors were on the following:

- AR1 values for multiple output problems
- ARMA11 values for multiple output problems
- Gaussian log likelihood values and derivatives for multiple output problems. Given that this is our workhorse distribution for many problems this was particularly concerning
- The ConstantModel which we use for testing previously had incorrect derivatives; these should always be 1 since f = theta -> df / dtheta = 1.

Covers #719

### CODE REVIEW

File: pints/toy/_constant_model.py [hunk 1]
```diff
@@ -106,8 +106,6 @@ def simulateS1(self, parameters, times):
             #   [df2/dp1, df2/dp2]]
             # i.e.
             #  [[1, 0],
-            #   [0, 2]]
-            # i.e. np.diag(self._r)
-            dy = np.tile(np.diag(self._r), (len(times), 1, 1))
+            #   [0, 1]]
+            dy = np.tile(np.diag(np.ones(len(self._r))), (len(times), 1, 1))
```
Comment (MichaelClerx) [inline pints/toy/_constant_model.py]: Thanks Ben!
Could just be `self._n` instead of `len(self._r)` then I think?

Comment (MichaelClerx) [reply pints/toy/_constant_model.py]: Main docstring needs updating too!

Comment (ben18785) [reply pints/toy/_constant_model.py]: @MichaelClerx Good idea. Both changed in my latest commit!

**Comment by ben18785** [comment]
@MichaelClerx Can we merge this please if you're happy with my changes?

**Pull request merged**